numpy/exceptions.py

```diff
@@ -0,0 +1,199 @@
+"""
+Exceptions and Warnings (:mod:`numpy.exceptions`)
+=================================================
+
+General exceptions used by NumPy.  Note that some exceptions may be module
+specific, such as linear algebra errors.
+
+.. versionadded:: NumPy 1.25
+
+    The exceptions module is new in NumPy 1.25.  Older exceptions remain
+    available through the main NumPy namespace for compatibility.
+
+.. currentmodule:: numpy.exceptions
+
+Warnings
+--------
+.. autosummary::
+   :toctree: generated/
+
+   ComplexWarning             Given when converting complex to real.
+   VisibleDeprecationWarning  Same as a DeprecationWarning, but more visible.
+
+Exceptions
+----------
+.. autosummary::
+   :toctree: generated/
+
+    AxisError       Given when an axis was invalid.
+    TooHardError    Error specific to `numpy.shares_memory`.
+
+"""
```

Comment:
Do we have autosummary elsewhere in a module docstring? This feels out of place to me.

Reply:
Hmmm, fft and polynomials used it.  But I am not sure the autosummary here is actually useful.

Reply:
Where?  Maybe @rossbar can just say.